### PR TITLE
Fix flaky WalletSpec property

### PR DESCRIPTION
**Pull request by rvl**
# Issue Number

Flaky test #2486

# Overview

Enhances assertion to prevent random failures.

### CODE REVIEW

**Review by jonathanknowles — APPROVED** [review]
LGTM.

I'm guessing that this would fail only very rarely, but good to fix nonetheless!

**Comment by jonathanknowles** [comment]
Oops:

```hs
Module ‘Test.QuickCheck’ does not export ‘(/==)’
```
I guess this should be `=/=`.

File: lib/core/test/unit/Cardano/WalletSpec.hs [hunk 1]
```diff
@@ -402,10 +403,10 @@ walletGetProp newWallet = monadicIO $ liftIO $ do
 walletGetWrongIdProp
     :: ((WalletId, WalletName, DummyState), WalletId)
     -> Property
-walletGetWrongIdProp (newWallet, corruptedWalletId) = monadicIO $ liftIO $ do
+walletGetWrongIdProp (newWallet@(wid, _, _), walletId) = monadicIO $ liftIO $ do
     (WalletLayerFixture _db wl _walletIds _) <- liftIO $ setupFixture newWallet
-    attempt <- runExceptT $ W.readWallet wl corruptedWalletId
-    attempt `shouldSatisfy` isLeft
+    attempt <- runExceptT $ W.readWallet wl walletId
+    attempt `shouldSatisfy` if wid /== walletId then isLeft else isRight
```
Comment (jonathanknowles) [inline lib/core/test/unit/Cardano/WalletSpec.hs]: I'm guessing this won't work, as `if` is expecting a `Bool` here, but the result of evaluating `=/=` is a `Property`.

Perhaps:
```suggestion
    attempt `shouldSatisfy` if wid /= walletId then isLeft else isRight
```

Comment (rvl) [reply lib/core/test/unit/Cardano/WalletSpec.hs]: oh yes, start compile & test, submit PR, then forget! fixed now.

**Comment by rvl** [comment]
Thanks.

> I'm guessing that this would fail only very rarely, but good to fix nonetheless!

Well, since it happened in a bors merge, and you opened the issue, and the problem is pretty clear, then here we are...

bors r+

**Pull request merged**